### PR TITLE
FAIMMS - faster data deletion

### PR DESCRIPTION
**Pull request by lbesnard**
* Add series of indexes on file_id on many tables. Daily deletion of files, and
   data from the DB, would take far too much time. This seems to be much quicker
   on local machine/subset of data.
 * create a delete data subjob for QAQC and another one for NO QAQC

### CODE REVIEW

File: workspace/FAIMMS/process/FAIMMS/subjobs/merged_qaqc_noqaqc_0.1.item [hunk 1]
```diff
@@ -141,7 +141,7 @@
...
...quot;TIME&quot; = measurements_noqaqc.&quot;TIME&quot; &#xA;               AND measurements_qaqc.&quot;VALUES&quot; = measurements_noqaqc.&quot;VALUES&quot;&#xA;               AND measurements_qaqc.channel_id = measurements_noqaqc.channel_id&#xA;    JOIN indexed_file table_a ON table_a.id = measurements_noqaqc.file_id&#xA;    JOIN timeseries_feature_deployments_varname table_b ON table_b.channel_id = measurements_noqaqc.channel_id&#xA;   ORDER BY measurements_noqaqc.channel_id, measurements_noqaqc.&quot;TIME&quot;;&#xA;&lt;/sql>&#xA;    &lt;/changeSet>&#xA;&#xA;&#xA;&#xA;&lt;changeSet author=&quot;talend&quot; id=&quot;create_indexes_for_faster_data_deletion&quot;>&#xA;  &lt;sql>&#xA;    CREATE INDEX file_id_idx &#x9;      ON indexed_file  USING btree  (id);&#xA;    CREATE INDEX file_id_noqaqc_idx   ON measurements_noqaqc  USING btree  (file_id);&#xA;    CREATE INDEX file_id_qaqc_idx     ON measurements_qaqc  USING btree  (file_id);&#xA;    CREATE INDEX file_id_var_idx      ON variable  USING btree  (file_id);&#xA;    CREATE INDEX file_id_att_idx      ON global_attributes_file  USING btree  (file_id);&#xA;&#xA; &lt;/sql>&#xA; &lt;/changeSet>&#xA;&#xA;&#xA;&lt;/databaseChangeLog>"/>
```
Comment (lbesnard) [inline workspace/FAIMMS/process/FAIMMS/subjobs/merged_qaqc_noqaqc_0.1.item]: main change here

**Comment by lbesnard** [comment]
will automatically merge on Friday if no review.

**Comment by danfruehauf** [comment]
Why so many new indexes?

**Comment by lbesnard** [comment]
when a file needs to be removed, the data needs to be removed from many different tables. so one index on file_id per table

**Comment by danfruehauf** [comment]
I'm just thinking whether this can be a foreign key, rather than need to index it a few times...

**Comment by lbesnard** [comment]
There are already some foreign keys on the measurement tables to link to the indexed_file table. But the measurement tables have something like 60 millions rows, so probably not a bad thing to add an index

**Comment by danfruehauf** [comment]
Please come and explain me when you have time...

**Comment by danfruehauf** [comment]
Verified with @jonescc - foreign keys will not imply searching on indexes in the original table, so adding those 3 indexes is fine.

**Pull request merged**